### PR TITLE
Grade User

### PR DESCRIPTION
**Pull request by maheshgudi**
made changes in the grade user with new UI
screenshot from 2016-04-18 15 04 38
screenshot from 2016-04-18 15 04 27
screenshot from 2016-04-18 15 04 11

### CODE REVIEW

File: yaksh/templates/yaksh/grade_user.html [hunk 1]
```diff
@@ -1,46 +1,102 @@
...
+{% endif %}
+</ul></tr>
+{% endfor %}
+</table>
+
+{% endif %}
+<div class="row">
+
+{%if users %}
+<div id = "student" class=span2>
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/grade_user.html]: "span2"

File: yaksh/templates/yaksh/grade_user.html [hunk 2]
```diff
@@ -1,46 +1,102 @@
...
+<br/>Questions correctly answered: {{ paper.get_answered_str }} <br/>
 Total attempts at questions: {{ paper.answers.count }} <br/>
-Marks obtained: {{ paper.get_total_marks }} <br/>
+Marks obtained: {{ paper.marks_obtained }} <br/>
 Start time: {{ paper.start_time }} <br/>
+{%if paper.percent%}
+Percentage obtained: {{paper.percent}}% <br/>
+{% endif %}
+{% if paper.passed == 0 %}
+Status : <b><font color = "red"> Failed </font></b><br/>
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/grade_user.html]: Please use a suitable class without this non-standard element.  At least use a style attribute!

File: yaksh/views.py [hunk 3]
```diff
@@ -853,34 +854,53 @@ def download_csv(request, questionpaper_id):
...
+    context = {"course_details": course_details}
+
+    if quiz_id != None:
+        questionpaper_id = QuestionPaper.objects.filter(quiz_id = quiz_id).values("id")
+
+        answerpaper_details = AnswerPaper.objects.filter(question_paper_id = questionpaper_id)
+        users_id = answerpaper_details.values_list("user_id").distinct()
+        user_details = User.objects.filter(id__in = users_id)
+        context = {"users": user_details, "quiz_id": quiz_id}
+        if user_id !=None:
```
Comment (prabhuramachandran) [inline yaksh/views.py]: Could this be refactored into a method on the model to provide the appropriate attempt when asked along with a test for the model?

File: yaksh/templates/yaksh/grade_user.html (hunk 2 above)
Comment (adityacp) [inline yaksh/templates/yaksh/grade_user.html]: instead of using <font> tag use css styling.

File: yaksh/templates/yaksh/grade_user.html (hunk 1 above)
Comment (maheshgudi) [inline yaksh/templates/yaksh/grade_user.html]: resolved

File: yaksh/templates/yaksh/grade_user.html (hunk 2 above)
Comment (maheshgudi) [inline yaksh/templates/yaksh/grade_user.html]: resolved

File: yaksh/views.py (hunk 3 above)
Comment (maheshgudi) [inline yaksh/views.py]: resolved

File: yaksh/models.py [hunk 4]
```diff
@@ -486,6 +486,15 @@ def get_user_answerpapers(self, user):
     def get_total_attempt(self, questionpaper, user):
         return self.filter(question_paper=questionpaper, user=user).count()
 
+    def get_users_for_questionpaper(self, questionpaper_id):
+        return self.get_answerpapers_for_quiz(questionpaper_id).values("user__id",
```
Comment (prabhuramachandran) [inline yaksh/models.py]: 80 cols please

File: yaksh/models.py [hunk 5]
```diff
@@ -486,6 +486,15 @@ def get_user_answerpapers(self, user):
...
+                                                                        "user__first_name",
+                                                                        "user__last_name"
+                                                                        ).distinct()
+    def get_user_all_attempts(self, questionpaper,user):
```
Comment (prabhuramachandran) [inline yaksh/models.py]: space between methods.

**Pull request closed without merging**